src/Password/Bcrypt.php

```diff
@@ -68,41 +67,35 @@ public function __construct($options = [])
      * Bcrypt
      *
      * @param  string $password
+     * @param  boolean $cryptuse
      * @throws Exception\RuntimeException
      * @return string
      */
-    public function create($password)
+    public function create($password, $cryptuse = false)
     {
         if (empty($this->salt)) {
             $salt = Rand::getBytes(self::MIN_SALT_SIZE);
         } else {
             $salt = $this->salt;
         }
-        $salt64 = mb_substr(str_replace('+', '.', base64_encode($salt)), 0, 22, '8bit');
-        /**
-         * Check for security flaw in the bcrypt implementation used by crypt()
-         * @see http://php.net/security/crypt_blowfish.php
-         */
-        $prefix = '$2y$';
-        $hash = crypt($password, $prefix . $this->cost . '$' . $salt64);
-        if (mb_strlen($hash, '8bit') < 13) {
-            throw new Exception\RuntimeException('Error during the bcrypt generation');
+
+        $options = [ 'cost' => $this->cost ];
+        if (PHP_VERSION_ID < 70000) { // salt is deprecated from PHP 7.0
+            $options['salt'] = $salt;
```

Comment:
Let's move lazy-generation of the salt into this block as well:

```php
$salt = $this->salt ?: Rand::getBytes(self::MIN_SALT_SIZE);
$options['salt'] = $salt;
```

If we never need to generate it (because PHP is doing it for us), let's not introduce a potentially expensive process!